### PR TITLE
chore(tests): unexport test helper functions from non `_test.go` files

### PR DESCRIPTION
**Pull request by qdm12**
## Changes

This is to reduce Go API noise and inter-package dependencies. Further PRs will be added to unexport these further (with other changes) and move them to `_test.go` files (not available in Go API).

- [x] Unexport test helper functions from non `_test.go` files, whenever possible (without other changes)

## Primary Reviewer

### CODE REVIEW

File: dot/utils.go [hunk 1]
```diff
@@ -24,8 +24,8 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-// NewTestGenesis returns a test genesis instance using "gssmr" raw data
-func NewTestGenesis(t *testing.T) *genesis.Genesis {
+// newTestGenesis returns a test genesis instance using "gssmr" raw data
+func newTestGenesis(t *testing.T) *genesis.Genesis {
```
Comment (EclesioMeloJunior) [inline dot/utils.go:28]: is it possible to move those not exported helpers test functions into `dot/utils_test.go` file? And move the function `RandomNodeName` into `dot/node.go` and then remove the file `./dot/utils.go`?

**Review by EclesioMeloJunior — CHANGES_REQUESTED** [review]
LGTM, just a small suggestion to remove the file `dot/utils.go`

File: dot/utils.go (hunk 1 above)
Comment (qdm12) [reply dot/utils.go:28]: That's definitely the longer term goal. We should have no test helper exported functions at all (prefer test helping code duplication vs package dependency), so the goal is to (through multiple PRs):

1. Unexport what's unexportable (this PR)
2. Unexport the remaining and duplicate test helper code in packages that need it (btw same goes for mocks in my opinion, we shouldn't share mocks across packages to reduce Go API noise).
3. Rename to utils_test.go (or helpers_test.go)

That 3-steps is to have the minimal amount of deltas too

**Review by noot — APPROVED** [review]

File: dot/utils.go (hunk 1 above)
Comment (EclesioMeloJunior) [reply dot/utils.go:28]: Got it!

**Review by EclesioMeloJunior — APPROVED** [review]

**Pull request merged**